lobby/src/main/java/org/triplea/lobby/server/db/UserController.java

```diff
@@ -57,11 +57,15 @@ public void updateUser(
         PreparedStatement ps =
             con.prepareStatement(
                 String.format(
-                    "update lobby_user set %s=?, email=? where username=?",
-                    getPasswordColumn(hashedPassword)))) {
+                    "update lobby_user set %s=?%s where username=?",
+                    getPasswordColumn(hashedPassword), email == null ? "" : ", email=?"))) {
       ps.setString(1, hashedPassword.value);
-      ps.setString(2, email);
-      ps.setString(3, name);
+      int index = 2;
+      if (email != null) {
+        ps.setString(index, email);
+        index++;
+      }
+      ps.setString(index, name);
```

Comment:
I personally find this very hard to read tbh.
I'd rather go with something like this:
```java
if (email != null) {
  ps.setString(2, email);
}
ps.setString(email != null ? 3 : 2, name);
```
Because IMO this makes it more clear which indices are used

Reply:
It's not terribly clean either way and the string format piece above is hacky. With just the two columns your suggestion is nicer, though I don't think it would be if there were more columns. 

I don't have a real preference on this either way, updated: 512746e0c